src/prefect/tasks.py

```diff
@@ -372,14 +389,7 @@ def __init__(
         if not hasattr(self.fn, "__qualname__"):
```

Comment:
This is a design nit, but this `not hasattr(...)` logic seems equally a part of the task key definition as the logic for when it does have the qualname attribute, so I suggest bringing this logic inside of the `_generate_task_key` function.

Reply:
good point! updated in [f8438c9](https://github.com/PrefectHQ/prefect/pull/14704/commits/f8438c95175a7db39ca8e6063a84c4184851338e)